### PR TITLE
feat: support PHP8.4, fixes #6205

### LINKED ISSUE #6205: Support PHP 8.4
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem?

PHP 8.4 should go into alpha before too long. It probably won't show up in deb.sury.org until a while after that. We'll probably want to provide some kind of support for it before it arrives in deb.sury.org, we've done this before in https://github.com/ddev/ddev-contrib and elsewhere.  Alpha1 is scheduled for June 6, 2024

### TODO

- [x] New ddev-php-base with 8.4 in it
- [x] Proper /etc/php from either ddev-webserver or ddev-php-base (choose which)

### Describe alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by rfay**
## The Issue

* #6205

## How This PR Solves The Issue

Updates to PHP 8.4alpha2

## TODO

- [x] Update default /etc/php in one place only. ddev-webserver/ddev-webserver-base-files/etc/php seems to have a full set of config files, but so does ddev-php-base/ddev-php-files/etc/php. They should be only one place, in ddev-php-base. This is confusing, and was already worked through once. In general, we want to get the defaults that the package will install.
- [x] PHP8.4 config files in /etc/php/8.4 should be mostly the defaults that it wants to install.
Related:
* https://github.com/ddev/ddev/pull/5219
* https://github.com/ddev/ddev/pull/5231
* [gist showing differences between ddev-php-base and ddev-webserver](https://gist.github.com/rfay/af74cde9dd9f1c6c865c3a4d754b28a3)

## Manual Testing Instructions

```
ddev config --php-version=8.4 && ddev start
ddev php --version
ddev php -i
```

## Automated Testing Overview

Mostly the standard testing applies.

Since some extensions like xhprof and xdebug are not available, there are short-circuits for those, that have to be removed later when they become available.

## Release/Deployment Notes

As PHP8.4 matures, the additional extensions will become available and have to be enabled in the Dockerfile. 

The normal set is 
> apcu bcmath bz2 curl cli common fpm gd imagick intl ldap mbstring memcached mysql opcache pgsql readline redis soap sqlite3 uploadprogress xdebug xhprof xml xmlrpc zip

Currently we have 
> bcmath bz2 cli common curl fpm gd intl ldap mbstring mysql opcache pgsql readline soap sqlite3 xml zip

Which means we're missing
> apcu opcache redis uploadprogress xdebug xhprof

### CODE REVIEW

File: docs/content/users/configuration/config.md [hunk 1]
```diff
@@ -440,7 +440,7 @@ The PHP version the project should use.
 
 | Type | Default | Usage
 | -- |---------| --
-| :octicons-file-directory-16: project | `8.2`   | Can be `5.6`, `7.0`, `7.1`, `7.2`, `7.3`, `7.4`, `8.0`, `8.1`, `8.2`, or `8.3`.
+| :octicons-file-directory-16: project | `8.2`   | Can be `5.6`, `7.0`, `7.1`, `7.2`, `7.3`, `7.4`, `8.0`, `8.1`, `8.2`, '8.3', or `8.3`.
```
Comment (rfay) [inline docs/content/users/configuration/config.md:443]:
```suggestion
| :octicons-file-directory-16: project | `8.2`   | Can be `5.6`, `7.0`, `7.1`, `7.2`, `7.3`, `7.4`, `8.0`, `8.1`, `8.2`, '8.3', or `8.4`.
```

File: containers/ddev-php-base/ddev-php-files/etc/php/8.4/fpm/php-fpm.conf [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+; Global Options ;
+;;;;;;;;;;;;;;;;;;
+
+[global]
+; Pid file
+; Note: the default prefix is /var
+; Default Value: none
+; Warning: if you change the value here, you need to modify systemd
+; service PIDFile= setting to match the value here.
+pid = /run/php/php8.3-fpm.pid
```
Comment (stasadev) [inline containers/ddev-php-base/ddev-php-files/etc/php/8.4/fpm/php-fpm.conf:19]:
```suggestion
pid = /run/php/php8.4-fpm.pid
```

File: containers/ddev-webserver/ddev-webserver-base-files/etc/apache2/conf-available/php8.4-fpm.conf [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    <FilesMatch ".+\.ph(?:ar|p|tml)$">
+        SetHandler "proxy:unix:/run/php-fpm.sock|fcgi://localhost"
+    </FilesMatch>
+# The default configuration works for most of the installation, however it could
+# be improved in various ways. One simple improvement is to not pass files that
+# doesn't exist to the handler as shown below, for more configuration examples
+# see https://wiki.apache.org/httpd/PHP-FPM
+#    <FilesMatch ".+\.ph(?:ar|p|tml)$">
+#        <If "-f %{REQUEST_FILENAME}">
+#            SetHandler "proxy:unix:/run/php/php8.3-fpm.sock|fcgi://localhost"
```
Comment (stasadev) [inline containers/ddev-webserver/ddev-webserver-base-files/etc/apache2/conf-available/php8.4-fpm.conf:20]:
```suggestion
#            SetHandler "proxy:unix:/run/php/php8.4-fpm.sock|fcgi://localhost"
```

File: pkg/ddevapp/schema.json [hunk 4]
```diff
@@ -274,7 +274,8 @@
         "8.0",
         "8.1",
         "8.2",
-        "8.3"
+        "8.3",
+        "8.4",
```
Comment (stasadev) [inline pkg/ddevapp/schema.json:278]:
```suggestion
        "8.4"
```

**Review by stasadev — APPROVED** [review]
Looks good to me. I only have small fixes.

**Comment by rfay** [comment]
Rebuilding php-base then:
- [ ] Rebuild ddev-webserver
- [ ] Run tests

**Pull request merged**